memory.go

```diff
@@ -36,20 +56,37 @@ type vmem interface {
 func deref[T any](r vmem, p ptr) T {
```

Comment:
Should we add a stricter constraint to the type parameter? `[T ptr32 | ptr64]` for example

Reply:
What do you have in mind?

Reply:
Maybe a constraint like:
```go
type dereferenceable interface {
  ~ptr32 | ~ptr64 | // etc...
}
```
I imagine there are types that would not be valid to pass (e.g. things containing inner pointers), so the compiler could help us validate that if we are more strict.

Reply:
I use `deref` on struct types. Is there a way to express a type constraint of structs only made of dereferenceable types?

otherwise the full type set is

```
type derefable interface {
	~int32 | ~int64 | ~uint32 | ~uint64 | ~byte | findfuncbucket | moduledata | functab | textsect | inlinedCall
}
```